### PR TITLE
[ConstraintSystem] [NFC] add tests for label matching diagnostics

### PR DESCRIPTION
**Pull request by omochi**
This patch add test cases for label matching diagnostics.

This haven't changes for compiler.

This is a part of following my other project.
https://github.com/apple/swift/pull/30444

I am planning to make many changes of diagnostics there particular corner cases.
It is useful to bake current behavior to repository as test cases 
before reviewing my that project.

### CODE REVIEW

**Comment by omochi** [comment]
@swift-ci please smoke test

File: test/Constraints/diagnostics.swift [hunk 1]
```diff
@@ -1415,6 +1415,107 @@ let f12: (Int) -> Void = { _ in }
 func f12<T : P2>(_ n: T, _ f: @escaping (T) -> T) {}
 f12(3, f4)// expected-error {{extra argument in call}}
 
+func testLabelErrorsBasic() {
```
Comment (LucianoPAlmeida) [inline test/Constraints/diagnostics.swift:1418]: I wonder if since there are quite a few tests specifically about label matching, is it a good idea to separate them in their own label matching file? This is just an idea because I see this `diagnostics` file is general, and maybe separate label matching diagnostics can make it easier to organize things :)

Comment (omochi) [reply test/Constraints/diagnostics.swift:1418]: I think it's a good idea.

Test cases about matching of call arguments are also in `test/Constraints/keyword_arguments.swift`.

I don't know how other authors distinguish them.

I choice `diagnostics.swift` for what has interest in unlabeled arguments because they has no *keyword*.

If we can make a big change,
I want to rename `keyword_arguments.swift` to `argument_matching.swift` and 
move some cases in `diagnostics.swift` to it.

Comment (omochi) [reply test/Constraints/diagnostics.swift:1418]: If we can, I want to split it more like:

- `argument_matching_labeled.swift`
- `argument_matching_unlabeled.swift`
- `argument_matching_variadic.swift`
- `argument_matching_defaulted.swift`
- `argument_matching_trailing_closure.swift`

Comment (LucianoPAlmeida) [reply test/Constraints/diagnostics.swift:1418]: I think since all fall in the context of `argument_matching` I think only one file is already fine.  
I don't know exactly what are the rules to this in the test suit, but to me, it seems reasonable to separate in, for example, an `argument_matching.swift` and have the subcases e.g. labeled, unlabeled, variadic.. divided inside the file.

Comment (xedin) [reply test/Constraints/diagnostics.swift:1418]: I think extracting all this into `test/Constraints/argument_matching.swift` would be just fine.

File: test/Sema/object_literals_osx.swift [hunk 2]
```diff
@@ -6,8 +6,11 @@ struct S: _ExpressibleByColorLiteral {
 }
 
 let y: S = #colorLiteral(red: 1, green: 0, blue: 0, alpha: 1)
-let y2 = #colorLiteral(red: 1, green: 0, blue: 0, alpha: 1) // expected-error{{could not infer type of color literal}} expected-note{{import AppKit to use 'NSColor' as the default color literal type}}
-let y3 = #colorLiteral(red: 1, bleen: 0, grue: 0, alpha: 1) // expected-error{{incorrect argument labels in call (have 'red:bleen:grue:alpha:', expected 'red:green:blue:alpha:')}} expected-error{{could not infer type of color literal}} expected-note{{import AppKit to use 'NSColor' as the default color literal type}}
+let y2 = #colorLiteral(red: 1, green: 0, blue: 0, alpha: 1) // expected-error{{could not infer type of color literal}}
```
Comment (xedin) [inline test/Sema/object_literals_osx.swift:9]: I think we should keep it as is since this changes is unrelated to other tests.

Comment (omochi) [reply test/Sema/object_literals_osx.swift:9]: OK.
I will move it to new another patch.

File: test/Constraints/diagnostics.swift (hunk 1 above)
Comment (omochi) [reply test/Constraints/diagnostics.swift:1418]: @LucianoPAlmeida @xedin Thanks to comment.
OK I will make new patch to create `argument_matching.swift`.

**Comment by omochi** [comment]
I will make two new other patches.

- (1) I make `argument_matching.swift` by renamed from `keyword_arguments.swift`.
  And move labeling cases in `diagnostics.swift` to it.

- (2) I break lines expectation in `object_lierals_osx.swift`.

After these works will be finished, I will adjust this patch to make it only add new tests.

**Comment by xedin** [comment]
Sounds like a plan!

**Comment by omochi** [comment]
@LucianoPAlmeida  @xedin I moved test cases to `argument_matching.swift`.

@swift-ci please smoke test

File: test/Constraints/argument_matching.swift [hunk 3]
```diff
@@ -1035,3 +1187,65 @@ CurriedClass.m1(2, b: 42)   // expected-error {{instance member 'm1' cannot be u
...
+
+  // 1 missing
+  f(0, 1, 2, dd: 3, ee: 4, ff: 5)
+  // expected-error@-1 {{missing argument label 'cc:' in call}} {{11-11=cc: }} {{none}}
+
+  // 1 extra
+  f(aa: 0, 1, cc: 2, dd: 3, ee: 4, ff: 5)
+  // expected-error@-1 {{extraneous argument label 'aa:' in call}} {{5-9=}} {{none}}
+
+  // 1 ooo
```
Comment (LucianoPAlmeida) [inline test/Constraints/argument_matching.swift:1210]: Just a question, what is 'ooo'?

Comment (xedin) [reply test/Constraints/argument_matching.swift:1210]: out of order

Comment (omochi) [reply test/Constraints/argument_matching.swift:1210]: ooo is out of order

Comment (LucianoPAlmeida) [reply test/Constraints/argument_matching.swift:1210]: Ah, thanks :)

**Comment by LucianoPAlmeida** [comment]
LGTM! @omochi :)

**Review by xedin — APPROVED** [review]
